card/lib/card/view/options.rb

```diff
@@ -176,16 +210,24 @@ def process_default_options
 
       # "foreign" options are non-standard options.  They're allowed, but they
       # prevent independent caching (and thus stubbing)
-      def foreign_options opts
-        opts.reject { |k, _v| Options.all_keys.include? k }
-      end
 
+      # non-standard options that are found in normalized_options
+      # @return [Hash] options Hash
       def foreign_normalized_options
-        @foreign_normalize_options ||= foreign_options normalized_options
+        @foreign_normalize_options ||= foreign_options_in normalized_options
       end
 
+      # non-standard options that are found in live_options
+      # @return [Hash] options Hash
       def foreign_live_options
-        foreign_options live_options
+        foreign_options_in live_options
+      end
+
+      # find non-standard option in Hash
+      # @param opts [Hash] options hash
+      # @return [Hash] options Hash
+      def foreign_options_in opts
+        opts.reject { |k, _v| Options.all_keys.include? k }
```

Comment:
How do I add foreign options?

Reply:
```
view :myview do |args|
  m = args[:my_foreign_arg]
end

render_myview my_foreign_arg: 4
```
...I'll add a coment to explain.

Reply:
I thought we want to get rid of args. 
And how do I change the type of the foreign args (like "heir" or "none")?

Reply:
Could be worth it to change the view api so that you have to define the foreign options that a view supports and requires.
For example `view :myview, optional: [:optional_arg], mandatory: [:mandatory_arg]` or just `view :myview do |mandatory_arg, optional_arg: nil| do`

Reply:
I did want to get rid of args, but the short version is that I couldn't quite yet; there are still too many places where we use them.

Another way to say this is that any thing that comes through in the |args| hash is a foreign option.  If it weren't foreign, it would be part of the voo API. You could justifiably argue that we could just add ways to access the foreign objects; I like your proposal to identify foreign options in the view definition as a way to make that much more compelling.

Really, I think there should be a ViewOption _class_ where we define these things, and the view definition should basically provide a means to define simple options on the fly.

I'll admit that I'm a little nervous to open this can of worms, but it sure would be nice to have a comprehensive solution in place...

Reply:
> And how do I change the type of the foreign args (like "heir" or "none")?

So far, that's not possible.  I think that's why you had to invent the `closest_live_option` thing. (foreign options can't be heirs).  Everything that's called on (at least within the gem) is a foreign_option: `followed_set`, `follow_option`, `follower`, and `inline_attachment_url`.

Perhaps some of those wouldn't be needed if they could be heirs?